projects/plugins/jetpack/extensions/blocks/blogroll/use-get-site-details.js

```diff
@@ -12,12 +12,21 @@ export default function useGetSiteDetails( { siteURL, subscriptions, enabled = f
 
 	if ( wpcomSite ) {
 		results.unshift( {
-			id: wpcomSite?.ID,
+			blog_id: wpcomSite?.ID,
 			description: wpcomSite?.description,
 			URL: wpcomSite?.URL,
 			site_icon: getSiteIcon( wpcomSite?.logo?.url ),
 			name: wpcomSite?.name,
 		} );
+	} else if ( validDomain && ! isLoading ) {
+		results.unshift( {
+			blog_id: validDomain,
```

Comment:
`blog_id: true` doesn't make sense to keep and track imo

Reply:
This happens because our API is returning `Facebook` as a wpcom site, which obviously is not. I could also see it happening with `gmail.com`. I am digging this API to understand why it happens, but it is for sure nothing related to blogroll only. Likewise, I believe whoever uses `https://public-api.wordpress.com/rest/v1.1/sites` API is exposed to it.

Reply:
We need to have a unique ID somehow, empty IDs won't work. Since these `non_wpcom_sites` have no ID, we thought we could use their URL, which is unique and the only information we have about it, since the ID is a string, it works fine.